From 9267a1e38df714b1881da7ad4b3e70d02cb48285 Mon Sep 17 00:00:00 2001
Message-Id: <9267a1e38df714b1881da7ad4b3e70d02cb48285.1424394676.git.feitong.yi@intel.com>
In-Reply-To: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
References: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Mon, 2 Feb 2015 14:42:13 +0530
Subject: [PATCH 08/61] MUST_REBASE [VPG]: drivers/video/adf: Enable DPMS
 on/off for DSI

As part of enabling encoders, driver does disable of PLL. Disabling of PLL
during enabling DSI encoder, does not allow DPMS on for DSI. This patch
fixes the DPMS ON/OFF calls for DSI.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

Issue: GMINL-5526
Change-Id: Idfb2f4d8263bc58f37581f4c553c9c0b16efcc13
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_pipeline.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_pipeline.c b/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
index 7aab631..6ee50cb 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
@@ -281,7 +281,8 @@ u32 vlv_pipeline_on(struct intel_pipeline *pipeline,
 
 	pr_info("ADF: %s: mode=%s\n", __func__, mode->name);
 
-	vlv_pll_disable(pll);
+	if (disp->type != INTEL_PIPE_DSI)
+		vlv_pll_disable(pll);
 	/* pll enable */
 	if (disp->type != INTEL_PIPE_DSI) {
 		if ((disp->type == INTEL_PIPE_DP) ||
-- 
1.7.9.5

